webapp/src/ts/modules/analytics/analytics-target-aggregates-sidebar-filter.component.ts

```diff
@@ -1,51 +1,47 @@
-import { Component, EventEmitter, OnDestroy, OnInit, Output } from '@angular/core';
+import { Component, EventEmitter, OnDestroy, OnInit, Output, Input } from '@angular/core';
 import { Subscription } from 'rxjs';
 import { Store } from '@ngrx/store';
 
 import { GlobalActions } from '@mm-actions/global';
 import { Selectors } from '@mm-selectors/index';
 import { ContactTypesService } from '@mm-services/contact-types.service';
 import { SettingsService } from '@mm-services/settings.service';
-import { UserSettingsService } from '@mm-services/user-settings.service';
 
 @Component({
   selector: 'mm-analytics-target-aggregates-sidebar-filter',
   templateUrl: './analytics-target-aggregates-sidebar-filter.component.html'
 })
 export class AnalyticsTargetAggregatesSidebarFilterComponent implements OnInit, OnDestroy {
 
+  @Input() userFacilities;
   @Output() facilitySelected = new EventEmitter<string>();
   private globalActions;
+  DEFAULT_FACILITY_LABEL = 'Facility';
   subscriptions: Subscription = new Subscription();
-  error;
   isOpen = false;
-  userFacilities;
+  error;
   selectedFacility;
-  selectedFacilityId;
   facilityFilterLabel;
 
   constructor(
     private store: Store,
     private contactTypesService: ContactTypesService,
     private settingsService: SettingsService,
-    private userSettingsService: UserSettingsService,
   ) {
     this.globalActions = new GlobalActions(store);
   }
 
   async ngOnInit() {
     try {
       this.subscribeToStore();
-      await this.loadUserFacility();
-      await this.setFacilityLabel();
+      this.facilityFilterLabel = await this.setFacilityLabel() || this.DEFAULT_FACILITY_LABEL;
```

Comment:
You don't have to change anything here, but I wanted to say that for the record I preferred having the `DEFAULT_FACILITY_LABEL` logic encapsulated down in `setFacilityLabel`.  I find it cleaner and easier to understand.  

(Also would not complain if you renamed the function to `getFacilityLabel`...)

Reply:
I wanted to set the default `DEFAULT_FACILITY_LABEL` once, but also I don't like it in the `ngOnInit`. 
I have reverted the change and left it in the set function.